src/adapters/poi_popup.js

```diff
@@ -0,0 +1,67 @@
+import {Popup} from 'mapbox-gl--ENV'
+import OsmSchedule from '../../src/adapters/osm_schedule'
+import Poi from '../mapbox/poi'
+import IconManager from "./icon_manager";
+const poiSubClass = require('../mapbox/poi_subclass')
+let popupTemplate = require('../views/popup.dot')
+const poiConfigs = require('../../config/constants.yml').pois
+
+const WAIT_BEFORE_DISPLAY = 800
+
+
+function PoiPopup() {}
+
+PoiPopup.prototype.init = function(map) {
+  this.map = map
+  this.popupHandle = null
+
+  this.timeOutHandler = null
+  this.isCounting = false
```

Comment:
This boolean `isCounting` can be removed, `timeOutHandler` is enough to implement the listener behaviour.